hm_pyhelper/util/sentry.py

```diff
@@ -0,0 +1,97 @@
+from typing import Dict, Union
+import hashlib
+
+EVENT_COUNT_KEY = "event_count"
+SUPPRESSION_COUNT_KEY = "suppression_count"
+
+# per session event counters, didn't feel the need to persist them
+event_counters = {}
+
+# maximum suppression interval. It grows exponentially till this value is reached.
+max_suppression_count = 128
+
+
+def sentry_fingerprint(hints: Dict) -> Union[str, None]:
+    """ return sha256 of log_record/exc_info as fingerprint"""
+    fingerprint_info_key = 'log_record'
+    if 'exc_info' in hints:
+        fingerprint_info_key = 'exc_info'
```

Comment:
@pritamghanghas - This seems to imply that same exception/error for different values will generate different finger prints? For example a `ValueError` for an expected number receiving values `not a number` or `really not a number` would create 2 different fingerprints right? as the resulting exception's info will contain these values too.

I think this is reasonable for now though as having the exc_info with values would give more insight into the error.

Reply:
@kashifpk we don't fingerprint exception's. I didn't see a point in trying to suppress uncaught exception. This is only for logging.error() with or without exc_info. I agree with different parameter value resulting in different fingerprint.  I didn't think of it. 

I can remove exc_info as log_record will also have it. But to make fingerprint cover "value differing cases", I think I should use only file and line number from LogRecord for better suppresion.
eg. fingerprint = f"{LogRecord.filename}:{LogRecord.lineno}:{LogRecord.funcName}"
basically not include anything about actual values in the fingerprint calculation.

Reply:
sentry also collects differing values together. Initially I had looked for a public api to get fingerprint from sentry itself. Didn't find it.

Reply:
update code to follow proposal